File: backend/components/metiers/threed/utils/rbac/samples/samples/examples/sample_advanced.js
// sample_advanced.js - Exemple métier avancé RBAC
function advancedSampleRBAC(user, action) {
  if (!user || !user.role) return false;
  if (user.role === 'admin') return true;
  if (user.role === 'user' && action === 'read') return true;
  return false;
}
module.exports = advancedSampleRBAC;

File: backend/components/metiers/threed/utils/rbac/samples/samples/examples/sample_advanced.py
# sample_advanced.py - Exemple métier avancé RBAC

def advanced_sample_rbac(user, action):
    if not user or 'role' not in user:
        return False
    if user['role'] == 'admin':
        return True
    if user['role'] == 'user' and action == 'read':
        return True
    return False
